### PR TITLE
Telegram Bot integration & Voice Prompt

### PR DESCRIPTION
**Pull request by Wladastic**
### Background
Adding 2 new prompt options:
 - Telegram bot (have to fix it, threading is wrong, it succeeds to send a message but the webhook won't work)
 - voice prompt (just realized it can work on WIndows and Linux too lol)

### Documentation

You have to create a telegram bot with botfather and get the chat_id after chatting with your bot the first time.

Once you set the bot up run:

To start run from AutoGPT folder :"python ./autogpt/start_bot.py --gpt3only"
or "python3 ./autogpt/start_bot.py --gpt3only"

it passes all the arguments into autogpt.

It will tell you through telegram to send "/start" to start it.
You can always use "/stop" to stop it and "/start" to start again.

I made the bot be able to restart your autogpt by this ;-)

### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [x] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [x] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

**Comment by nponeccop** [comment]
@Wladastic There are conflicts.

**Comment by nponeccop** [comment]
If it doesn't work yet GitHub lets you to convert a PR to a draft PR

File: voices/voices.sh
Comment (nponeccop) [inline voices/voices.sh:1]: Don't forget to revert

**Comment by Wladastic** [comment]
@nponeccop  Feel free to test.
It's kindda flacky still but I probably need some help fixing the thread blocking by waiting for the users's response.
It works, but sometimes just gets stuck at the responce_received.wait()

**Comment by Wladastic** [comment]
image

**Comment by nponeccop** [comment]
@Wladastic There are conflicts and red CI now

**Comment by nponeccop** [comment]
@Wladastic The CI is still red (just in case you didn't notice)

**Comment by Wladastic** [comment]
I know, only updated.I found a better was to implement it with the new structure of the project but have to fix some stuff, as it still get’s stuch at response.wait() and doesn’t let the application poll do an update, I didn’t do much python before hahaAny ideas?Am 15.04.2023 um 18:29 schrieb nponeccop ***@***.***>:﻿
@Wladastic The CI is still red (just in case you didn't notice)

**Comment by Wladastic** [comment]
> @Wladastic The CI is still red (just in case you didn't notice)

It now works!
Waiting for the pipelines to finish so that I can see what whitespaces I missed, then it's ready to be merged! :)

**Comment by Wladastic** [comment]
image

**Comment by Wladastic** [comment]
@richbeales  if this is marked plugin, will it be skipped or what does it mean?

**Review by nponeccop — DISMISSED** [review]

**Comment by Wladastic** [comment]
@nponeccop will need your approve again, there have been many changes on master and I had to test and fix some stuff.
Also I rebuilt the /start /stop handlers, you can now just /stop the bot and /start again without having to touch the console ever :D

**Review by nponeccop — DISMISSED** [review]

**Comment by Wladastic** [comment]
@nponeccop sorry mate, had to fix the conflicts due to some merges.

**Comment by Wladastic** [comment]
No idea what black does?!

**Comment by Pwuts** [comment]
Like #2593, this is functionality that fits better in a plugin than in the core. If you need help converting it to a plugin, [at]BillSchumacher is happy to help. :)

**Comment by Wladastic** [comment]
Yup we already talk. 
only issue is wrapping autogpt with the bot as I added a possibility to restart autogpt by telegram

**Comment by Wladastic** [comment]
I started making a plugin here:
https://github.com/Wladastic/Auto-GPT-Telegram-Plugin
If anyone likes to assist, I would be glad :)

**Pull request closed without merging**